main.go

```diff
@@ -141,7 +142,22 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := utils.InitOutOfServiceTaintFlags(mgr.GetConfig()); err != nil {
+	interval := 2 * time.Second // retry every 2 seconds
+	timeout := 10 * time.Second // for a period of 10 seconds
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	// Using wait.PollUntilContextTimeout to retry InitOutOfServiceTaintFlags in case there is a temporary network issue.
+	// Since the last internal error returned by InitOutOfServiceTaintFlags also indicates whether polling succeed or not, there is no need to also keep the context error returned by PollUntilContextTimeout.
+	_ = wait.PollUntilContextTimeout(ctx, interval, timeout, true, func(ctx context.Context) (bool, error) {
+		if err = utils.InitOutOfServiceTaintFlags(mgr.GetConfig()); err != nil {
```

Comment:
I understand that this is unlikely, however, IIUC, the poll might retry even if the error is
returned by `setOutOfTaintFlags`, which is not what we intend, isn'it?

I would suggest to apply the retry only to the part that gets the version from the cluster, either
moving it inside `InitOutOfServiceTaintFlags`, or rework the function into two separate parts
- first get the version (with retry)
- second call `setOutOfTaintFlags`

Reply:
> IIUC, the poll might retry even if the error is returned by setOutOfTaintFlags

I agree, but I don't think it's a problem.

Since InitOutOfServiceTaintFlags failing isn't a blocker I think it's fine to retry the whole method.
  - Best case, we will recover other use cases we haven't considered (another very unlikely use case is: we'll get a corrupted version due to network issue which will cause  setOutOfTaintFlags to fail and retrying the whole method will fix that )
  - Worst case, we'll run some unnecessary retries during SNR setup which will delay the setup by couple of seconds